<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Sebastian Goth &lt;seezer@roath.org&gt;</author>
 <class>IgnoreListEditDlg</class>
 <widget class="QDialog" name="IgnoreListEditDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>389</width>
    <height>368</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Configure Ignore Rule</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/icons/oxygen/16x16/actions/configure.png</normaloff>:/icons/oxygen/16x16/actions/configure.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QGroupBox" name="strictnessGroupBox">
       <property name="toolTip">
        <string>&lt;p&gt;&lt;b&gt;Strictness:&lt;/b&gt;&lt;/p&gt;
&lt;p&gt;&lt;u&gt;Dynamic:&lt;/u&gt;&lt;/p&gt;
&lt;p&gt;Messages are filtered &quot;on the fly&quot;.
Whenever you disable/delete the ignore rule, the messages are shown again.&lt;/p&gt;
&lt;p&gt;&lt;u&gt;Permanent:&lt;/u&gt;&lt;/p&gt;
&lt;p&gt;Messages are filtered before they get stored in the database.&lt;/p&gt;</string>
       </property>
       <property name="title">
        <string>Strictness</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QRadioButton" name="dynamicStrictnessButton">
          <property name="text">
           <string>Dynamic</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="permanentStrictnessButton">
          <property name="text">
           <string>Permanent</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="typeGroupBox">
       <property name="toolTip">
        <string>&lt;p&gt;&lt;b&gt;Rule Type:&lt;/b&gt;&lt;/p&gt;
&lt;p&gt;&lt;u&gt;Sender:&lt;/u&gt;&lt;/p&gt;
&lt;p&gt;The rule is matched against the sender string
&lt;i&gt;nick!ident@host.name&lt;i&gt;&lt;/p&gt;
&lt;p&gt;&lt;u&gt;Message:&lt;/u&gt;&lt;/p&gt;
&lt;p&gt;The rule is matched against the actual message content&lt;/p&gt;</string>
       </property>
       <property name="title">
        <string>Rule Type</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QRadioButton" name="senderTypeButton">
          <property name="text">
           <string>Sender</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="messageTypeButton">
          <property name="text">
           <string>Message</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="ctcpTypeButton">
          <property name="text">
           <string>CTCP</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="ignoreRuleGroupBox">
     <property name="toolTip">
      <string>&lt;p&gt;&lt;b&gt;Ignore rule:&lt;/b&gt;&lt;/p&gt;
&lt;p&gt;Depending on the type of the rule, the text is matched against either:&lt;/p&gt;
&lt;p&gt;&lt;u&gt;- the message content:&lt;/u&gt;&lt;/p&gt;
&lt;p&gt;&lt;i&gt;Example:&lt;/i&gt;
&lt;br /&gt;
&lt;i&gt;*foobar*&lt;/i&gt; matches any text containing the word &lt;i&gt;foobar&lt;/i&gt;&lt;/p&gt;
&lt;p&gt;&lt;u&gt;- the sender string&lt;/u&gt;  &lt;i&gt;(nick!ident@host.name)&lt;/i&gt;&lt;/p&gt;
&lt;p&gt;&lt;i&gt;Examples:&lt;/i&gt;
&lt;br /&gt;
- &lt;i&gt;*@foobar.com&lt;/i&gt; matches any sender from host &lt;i&gt;foobar.com&lt;/i&gt;
&lt;br /&gt;
- &lt;i&gt;stupid!.+&lt;/i&gt; (RegEx) matches any sender with nickname &lt;i&gt;stupid&lt;/i&gt; from any host&lt;/p&gt;</string>
     </property>
     <property name="title">
      <string>Ignore Rule</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLineEdit" name="ignoreRuleLineEdit"/>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QCheckBox" name="isRegExCheckBox">
          <property name="toolTip">
           <string>&lt;p&gt;&lt;b&gt;Use regular expressions:&lt;/b&gt;&lt;/p&gt;
&lt;p&gt;If enabled, rules follow regular expression syntax.&lt;/p&gt;
&lt;p&gt;Otherwise rules allow wildcard matching with the following special characters:&lt;/p&gt;
&lt;p&gt; *: representing &quot;any amount of any character&quot;
&lt;br /&gt;
?: representing &quot;exactly one character&quot;&lt;/p&gt;</string>
          </property>
          <property name="text">
           <string>Regular expression</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="scopeGroupBox">
     <property name="toolTip">
      <string>&lt;p&gt;&lt;b&gt;Scope:&lt;/b&gt;&lt;/p&gt;
&lt;p&gt;&lt;u&gt;Global:&lt;/u&gt;&lt;/p&gt;
&lt;p&gt;The rule is active for any channel on any network&lt;/p&gt;
&lt;p&gt;&lt;u&gt;Network:&lt;/u&gt;&lt;/p&gt;
&lt;p&gt;The list below is interpreted as a list of networks for which the rule should match&lt;/p&gt;
&lt;p&gt;&lt;u&gt;Channel:&lt;/u&gt;&lt;/p&gt;
&lt;p&gt;The list below is interpreted as a list of channels for which the rule should match&lt;/p&gt;</string>
     </property>
     <property name="title">
      <string>Scope</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QRadioButton" name="globalScopeButton">
          <property name="text">
           <string>Global</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="networkScopeButton">
          <property name="text">
           <string>Network</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="channelScopeButton">
          <property name="text">
           <string>Channel</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="scopeRuleTextEdit">
        <property name="toolTip">
         <string>&lt;p&gt;&lt;b&gt;Scope rule:&lt;/b&gt;&lt;/p&gt;
&lt;p&gt;A scope rule is a semicolon separated list of either &lt;i&gt;network&lt;/i&gt; or &lt;i&gt;channel&lt;/i&gt; names.&lt;/p&gt;
&lt;p&gt;&lt;i&gt;Example:&lt;/i&gt;
&lt;br /&gt;
&lt;i&gt;#quassel*; #foobar; !#quasseldroid&lt;/i&gt;
&lt;br /&gt;
would match on &lt;i&gt;#foobar&lt;/i&gt; and any channel starting with &lt;i&gt;#quassel&lt;/i&gt; except for &lt;i&gt;#quasseldroid&lt;/i&gt;
&lt;br /&gt;
&lt;p&gt;If only inverted names are specified, it will match anything except for what's specified (implicit wildcard).&lt;/p&gt;
&lt;p&gt;&lt;i&gt;Example:&lt;/i&gt;
&lt;br /&gt;
&lt;i&gt;!#quassel*; !#foobar&lt;/i&gt;
&lt;br /&gt;
would match anything except for &lt;i&gt;#foobar&lt;/i&gt; or any channel starting with &lt;i&gt;#quassel&lt;/i&gt;&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QCheckBox" name="isActiveCheckBox">
       <property name="toolTip">
        <string>&lt;p&gt;&lt;b&gt;Enable / Disable:&lt;/b&gt;&lt;/p&gt;
&lt;p&gt;Only enabled rules are filtered.
&lt;br /&gt;
For dynamic rules, disabling actually shows the filtered messages again.&lt;/p&gt;</string>
       </property>
       <property name="text">
        <string>Rule is enabled</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>dynamicStrictnessButton</tabstop>
  <tabstop>permanentStrictnessButton</tabstop>
  <tabstop>senderTypeButton</tabstop>
  <tabstop>messageTypeButton</tabstop>
  <tabstop>ignoreRuleLineEdit</tabstop>
  <tabstop>isRegExCheckBox</tabstop>
  <tabstop>scopeRuleTextEdit</tabstop>
  <tabstop>globalScopeButton</tabstop>
  <tabstop>networkScopeButton</tabstop>
  <tabstop>channelScopeButton</tabstop>
  <tabstop>isActiveCheckBox</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources>
  <include location="../../../icons/oxygen_icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>IgnoreListEditDlg</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>IgnoreListEditDlg</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
